WordPress/Classes/Models/Suggestion.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+@objcMembers public class Suggestion: NSObject {
+    let userLogin: String?
+    let displayName: String?
+    let imageURL: URL?
+
+    init(dictionary: [String: Any]) {
+        userLogin = dictionary["user_login"] as? String
+        displayName = dictionary["display_name"] as? String
+        if let imageURLString = dictionary["image_URL"] as? String {
+            imageURL = URL(string: imageURLString)
+        } else {
+            imageURL = nil
+        }
+    }
+
+    public func cachedAvatar(with size: CGSize) -> UIImage? {
+        var hash: NSString?
+        let type = avatarSourceType(with: &hash)
+
+        if let hash = hash, let type = type {
+            return WPAvatarSource.shared()?.cachedImage(forAvatarHash: hash as String, of: type, with: size)
+        }
+        return nil
+    }
+
+    public func fetchAvatar(with size: CGSize, success: ((UIImage?) -> Void)?) {
+        var hash: NSString?
+        let type = avatarSourceType(with: &hash)
+
+        if let hash = hash, let type = type, let success = success {
+            WPAvatarSource.shared()?.fetchImage(forAvatarHash: hash as String, of: type, with: size, success: success)
+        } else {
+            success?(nil)
+        }
+    }
+
+    func avatarSourceType(with hash: inout NSString?) -> WPAvatarSourceType? {
```

Comment:
I'd love it if we could update `WPAvatarSource.parseURL(forAvatarHash:)` so that it doesn't require an inout parameter, but I guess that might be outside of the scope of these changes.

Reply:
That's an idea. I'll take a look and see if it's easy, if not I'll probably leave it out.

Reply:
I think it's best to address it separately, either by porting `WPAvatarSource` to Swift or making its API more Swift-friendly. I created a ticket for me to follow up on that: https://github.com/wordpress-mobile/WordPress-iOS/issues/14996